datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1079,57 +1317,178 @@ impl LogicalPlan {
     }
 }
 
+/// This macro is used to determine continuation during combined transforming
```

Comment:
I think this may be dead code -- can somone check https://github.com/apache/arrow-datafusion/pull/9997 ?

Reply:
Actually, I deliberately added `..._with_subqueries()` versions of all base `TreeNode` APIs to `LogicalPlan`. These 3 macros are used in `LogicalPlan::rewrite_with_subqueries()`, `LogicalPlan::transform_down_with_subqueries()` and `LogicalPlan::transform_up_with_subqueries()`.

Reply:
Cool -- thanks -- I think that will be clearer as we consolidate the tree rewriting code